packages/shared/src/server/queries/createTracesQuery.ts

```diff
@@ -0,0 +1,87 @@
+import { Prisma } from "@prisma/client";
+
+export function createTracesQuery({
+  select,
+  projectId,
+  observationTimeseriesFilter = Prisma.empty,
+  page,
+  limit,
+  searchCondition = Prisma.empty,
+  filterCondition = Prisma.empty,
+  orderByCondition = Prisma.empty,
+  selectScoreValues = false,
+}: {
+  select: Prisma.Sql;
+  projectId: string;
+  observationTimeseriesFilter?: Prisma.Sql;
+  page?: number;
+  limit?: number;
+  searchCondition?: Prisma.Sql;
+  filterCondition?: Prisma.Sql;
+  orderByCondition?: Prisma.Sql;
+  selectScoreValues?: boolean;
+}) {
+  return Prisma.sql`
+  SELECT
+      ${select}
+  FROM
+    "traces" AS t
+  LEFT JOIN LATERAL (
+    SELECT
+      SUM(prompt_tokens) AS "promptTokens",
+      SUM(completion_tokens) AS "completionTokens",
+      SUM(total_tokens) AS "totalTokens",
+      SUM(calculated_total_cost) AS "calculatedTotalCost",
+      SUM(calculated_input_cost) AS "calculatedInputCost",
+      SUM(calculated_output_cost) AS "calculatedOutputCost",
+      COALESCE(  
+        MAX(CASE WHEN level = 'ERROR' THEN 'ERROR' END),  
+        MAX(CASE WHEN level = 'WARNING' THEN 'WARNING' END),  
+        MAX(CASE WHEN level = 'DEFAULT' THEN 'DEFAULT' END),  
+        'DEBUG'  
+      ) AS "level"
+    FROM
+      "observations_view"
+    WHERE
+      trace_id = t.id
+      AND "type" = 'GENERATION'
+      AND "project_id" = ${projectId}
+      ${observationTimeseriesFilter}
+  ) AS tm ON true
+  LEFT JOIN LATERAL (
+    SELECT
+      COUNT(*) AS "observationCount",
+      EXTRACT(EPOCH FROM COALESCE(MAX("end_time"), MAX("start_time"))) - EXTRACT(EPOCH FROM MIN("start_time"))::double precision AS "latency"
+    FROM
+        "observations"
+    WHERE
+        trace_id = t.id
+        AND "project_id" = ${projectId}
+         ${observationTimeseriesFilter}
+  ) AS tl ON true
+  LEFT JOIN LATERAL (
+    SELECT
+      ${selectScoreValues ? Prisma.sql`jsonb_object_agg(name::text, "values") AS "scores_values",` : Prisma.empty}
+      jsonb_object_agg(name::text, avg_value::double precision) AS "scores_avg"
+    FROM (
+        SELECT
+            name,
+            ${selectScoreValues ? Prisma.sql`array_agg(COALESCE(string_value, value::text)) AS "values",` : Prisma.empty}
```

Comment:
Note: the numeric score values are cast as string and show up in a format like `["0.8042602923476814","0.7271858775794038","-0.2272392299725476","0.9284890924655911"]` in the csv or json output example -- while this is not ideal I think it's fine for the time being and can always be improved later, wdyt?

Reply:
Yes, agree :)